src/unit_test.c

```diff
@@ -0,0 +1,408 @@
+#if !defined(_PID_T) && !defined(pid_t)
+#   include <sys/types.h>
+#endif
+#    define SUPPORTS_CONCURRENCY 1
+#  endif
+#endif
+#ifndef SUPPORTS_CONCURRENCY
+#  define SUPPORTS_CONCURRENCY 0
+#endif
+
+#include "unit_test.h"
+#include "testrand.h"
+#include "tests_common.h"
+
+/* Number of times certain tests will run */
+int COUNT = 16;
+
+static int parse_jobs_count(const char* key, const char* value, struct Args* out);
+static int parse_iterations(const char* arg);
+static int parse_target(const char* value, struct TestFramework* tf);
+
+/*
+ *   Main entry point for handling command-line arguments.
+ *
+ *   This function is responsible for parsing a single key/value pair
+ *   (e.g., -jobs=4) and updating the provided Args struct accordingly.
+ *
+ *   Developers should extend this function whenever new command-line
+ *   options are introduced. Each new argument should be validated,
+ *   converted to the appropriate type, and stored in the 'Args' struct.
+ */
+static int parse_arg(const char* key, const char* value, struct TestFramework* tf) {
+    /* Number of concurrent tasks */
+    if (strcmp(key, "j") == 0 || strcmp(key, "jobs") == 0) {
+        return parse_jobs_count(key, value, &tf->args);
+    }
+    /* Number of iterations */
+    if (strcmp(key, "iter") == 0 || strcmp(key, "iterations") == 0) {
+        return parse_iterations(value);
+    }
+    /* Custom seed */
+    if (strcmp(key, "seed") == 0) {
+        tf->args.custom_seed = (!value || strcmp(value, "NULL") == 0) ? NULL : value;
+        return 0;
+    }
+    /* Test target */
+    if (strcmp(key, "t") == 0 || strcmp(key, "target") == 0) {
+        return parse_target(value, tf);
+    }
+    /* Logging */
+    if (strcmp(key, "log") == 0) {
+        tf->args.logging = value && strcmp(value, "1") == 0;
+        return 0;
+    }
+
+    /* Unknown key: report just so typos don’t silently pass. */
+    printf("Unknown argument '-%s=%s'\n", key, value);
+    return 0;
+}
+
+static void help(void) {
+    printf("Usage: ./tests [options]\n\n");
+    printf("Run the test suite for the project with optional configuration.\n\n");
+    printf("Options:\n");
+    printf("    -help, -h                           Show this help message\n");
+    printf("    -print_tests                        Display list of all available tests and modules\n");
+    printf("    -j=<num>, -jobs=<num>               Number of parallel worker processes (default: 0 = sequential)\n");
+    printf("    -iter=<num>, -iterations=<num>      Number of iterations for each test (default: 16)\n");
+    printf("    -seed=<hex>                         Set a specific RNG seed (default: random)\n");
+    printf("    -target=<test name>, -t=<name>      Run a specific test (can be provided multiple times)\n");
+    printf("    -target=<module name>, -t=<module>  Run all tests within a specific module (can be provided multiple times)\n");
```

Comment:
Is there a specific reason to overload the `-target` option?  If not, could a different one be used instead?

Reply:
> Is there a specific reason to overload the `-target` option? If not, could a different one be used instead?

Mainly because both are targets; the only diff is that one target runs more test cases than the other one.
In general, it’s similar to how most unit test frameworks work. For example, on `boost` and `gtest` you can run groups of tests and single tests with the same command (`--run_test` and `--gtest_filter` respectively); the only diff with them is that they add the module name as a prefix when specifying single test case (which we could also do here).

Reply:
Right. Let's keep it this work. It works nicely with CMake code. See https://github.com/hebasto/secp256k1/commit/3ff3105edc5e78aaceae0dddba5f788ec83a163d.